registry/proxy/proxymetrics.go

```diff
@@ -3,6 +3,22 @@ package proxy
 import (
 	"expvar"
 	"sync/atomic"
+
+	prometheus "github.com/distribution/distribution/v3/metrics"
+	"github.com/docker/go-metrics"
+)
+
+var (
+	// requests is the number of total incoming proxy request received for blob/manifest
+	requests = prometheus.ProxyNamespace.NewLabeledCounter("requests", "The number of total incoming proxy request received", "type")
+	// hits is the number of total proxy request hits for blob/manifest
+	hits = prometheus.ProxyNamespace.NewLabeledCounter("hits", "The number of total proxy request hits", "type")
+	// hits is the number of total proxy request misses for blob/manifest
+	misses = prometheus.ProxyNamespace.NewLabeledCounter("misses", "The number of total proxy request misses", "type")
+	// bytesPulled is the size of total bytes pulled from the upstream for blob/manifest
+	bytesPulled = prometheus.ProxyNamespace.NewLabeledCounter("bytes_pulled", "The size of total bytes pulled from the upstream", "type")
+	// bytesPushed is the size of total bytes pushed to the client for blob/manifest
+	bytesPushed = prometheus.ProxyNamespace.NewLabeledCounter("bytes_pushed", "The size of total bytes pushed to the client", "type")
```

Comment:
Silly question (I haven't used prometheus in a long time); given that we're using prefixes, I'm assuming it's possible to filter based on prefix; in that case, would it be beneficial to swap `pulled/pushed` and `bytes`. In that case it would be possible to filter on `registry_proxy_pulled` and `registry_proxy_pushed` prefixes (in case other information gets added related to `pushed` or `pulled` in future;

- `registry_proxy_bytes_pulled_total` -> `registry_proxy_pulled_bytes_total`
- `registry_proxy_bytes_pushed_total` -> `registry_proxy_pushed_bytes_total`